### ISSUE TITLE
Unable to search or filter on Enum values

### ISSUE THREAD

**Issue by malcomm** [aws-amplify/amplify-js#4392]
** Which Category is your question related to? **
GraphQL API

** What AWS Services are you utilizing? **
Appsync / Amplify CLI / GraphQL

** Provide additional details e.g. code snippets **

I would like to be able to search for Enum values and in this specific use case, I would like to search for all Responses that do not have the status of `draft` and `verified`. I don't see a way to do this using the generated code for base List function, or trying to use the `@key` or using `@searchable`.

Sample excerpt schema:
```
type Response
  @model
  @key(name: "ResponseStatus", fields: ["status", "updatedAt"], queryField: "responseStatus")
  @searchable
{
  id: ID!,
  status: Status,
  updatedAt: AWSDate
  ...
}

enum Status {
  draft
  submitted
  reviewed
  modified
  resubmitted
  verified
  inactive
  unknown
}
```

At this point, the only option I see moving forward is to change this enum to a String ... which is really frustrating.

Basically, I'm asking for first level support of enums. Any help?
